### PR TITLE
kucoin implement fetchLedger fix #4880

### LINKED ISSUE #4880: Implement fetchLedger on kucoin
- CCXT version: 1.18.395
- Exchange: kucoin
- Method: fetchLedger

Just saw that kucoin now also has an endpoint for getting the ledger entries which is much more convenient than fetching trades/withdrawals/deposits separately. As some of your other exchange implementations (e.g. kraken) already have the fetchLedger method, might be supercool if you could also implement it for kucoin :-)

### PR DESCRIPTION
**Pull request by frosty00**
it's sunday 0:

### CODE REVIEW

File: js/kucoin.js [hunk 1]
```diff
@@ -1329,6 +1329,154 @@ module.exports = class kucoin extends Exchange {
...
+        //         amount: '0.0111123',
+        //         bizType: 'Exchange',
+        //         balance: '0.13224427',
+        //         fee: '0.0000111',
+        //         context: '{"symbol":"KCS-ETH","orderId":"5d18ab98c788c6426188296f","tradeId":"5d18ab9818996813f539a806"}',
+        //         currency: 'ETH',
+        //         direction: 'out'
+        //     }
+        //
+        // withdrawal
```
Comment (kroitor) [inline js/kucoin.js]: No deposits?

Comment (frosty00) [reply js/kucoin.js]: yep, the format is the same: https://github.com/ccxt/ccxt/pull/5401/files#diff-03145d38fecb5a74e76f74154826789dR1474

File: js/kucoin.js [hunk 2]
```diff
@@ -1329,6 +1329,154 @@ module.exports = class kucoin extends Exchange {
...
+        const typeId = this.safeString (item, 'bizType');
+        const type = this.parseLedgerEntryType (typeId);
+        const contextString = this.safeString (item, 'context');
+        let id = undefined;
+        if (this.isJsonEncodedObject (contextString)) {
+            const context = this.parseJson (contextString);
+            if (type === 'trade') {
+                id = this.safeString (context, 'tradeId');
+            } else if (type === 'transaction') {
+                id = this.safeString (context, 'txId');
```
Comment (kroitor) [inline js/kucoin.js]: These values go to `refId`, and the `orderId` goes to `id`.

Comment (frosty00) [reply js/kucoin.js]: could be more clear about it here - will add necessary edits

**Pull request merged**